motionplan/ik/nlopt.go

```diff
@@ -178,19 +179,26 @@ func (ik *nloptIK) Solve(ctx context.Context,
 		} else if solutionRaw == nil {
 			panic("why is solutionRaw nil")
 		} else if result < defaultGoalThreshold || !ik.exact {
-			solutionChan <- &Solution{
+			solution := &Solution{
 				Configuration: solutionRaw,
 				Score:         result,
 				Exact:         result < defaultGoalThreshold,
 			}
-			solutionsFound++
+			select {
+			case <-ctx.Done():
```

Comment:
The old code saying "we need to drain the solution channel" was because of this missing select on `ctx.Done()` resulting in a deadlock.